File: sdk/android/story-sdk/src/main/kotlin/com/openstory/sdk/internal/network/SdkFeedModels.kt
package com.openstory.sdk.internal.network

internal data class SdkFeedRequestPayload(
    val clientId: String,
    val placementKey: String,
    val platform: String,
    val appVersion: String,
    val userSegments: List<String>,
)

internal data class SdkFeedResponsePayload(
    val clientId: String,
    val placementKey: String,
    val context: SdkFeedContextPayload,
    val resolvedSet: SdkFeedSetPayload? = null,
    val generatedAt: String,
)

internal data class SdkFeedContextPayload(
    val platform: String,
    val appVersion: String,
    val userSegments: List<String>,
)

internal data class SdkFeedSetPayload(
    val id: String,
    val revisionId: String,
    val placementKey: String,
    val isFallback: Boolean,
    val groups: List<SdkFeedGroupPayload>,
)

internal data class SdkFeedGroupPayload(
    val id: String,
    val revisionId: String,
    val title: String,
    val bottomLabel: String? = null,
    val logoUrl: String,
    val badge: SdkFeedBadgePayload? = null,
    val stories: List<SdkFeedStoryPayload>,
)

internal data class SdkFeedBadgePayload(
    val type: String,
    val value: String,
)

internal data class SdkFeedStoryPayload(
    val id: String,
    val revisionId: String,
    val title: String,
    val mediaType: String,
    val imageDurationMs: Long? = null,
    val asset: SdkFeedAssetPayload,
    val posterAsset: SdkFeedAssetPayload? = null,
    val cta: SdkFeedCtaPayload? = null,
)

internal data class SdkFeedAssetPayload(
    val id: String,
    val url: String,
    val mimeType: String,
    val width: Int? = null,
    val height: Int? = null,
    val durationMs: Long? = null,
)

internal data class SdkFeedCtaPayload(
    val label: String,
    val type: String,
    val value: String,
)

File: sdk/flutter/ios/open_story_flutter/Sources/open_story_flutter/OpenStorySDK/Internal/Network/SdkFeedModels.swift
import Foundation

internal struct SdkFeedRequestPayload: Codable, Equatable, Sendable {
    let clientId: String
    let placementKey: String
    let platform: String
    let appVersion: String
    let userSegments: [String]

    enum CodingKeys: String, CodingKey {
        case clientId = "client_id"
        case placementKey = "placement_key"
        case platform
        case appVersion = "app_version"
        case userSegments = "user_segments"
    }
}

internal struct SdkFeedResponsePayload: Codable, Equatable, Sendable {
    let clientId: String
    let placementKey: String
    let context: SdkFeedContextPayload
    let resolvedSet: SdkFeedSetPayload?
    let generatedAt: String

    enum CodingKeys: String, CodingKey {
        case clientId = "client_id"
        case placementKey = "placement_key"
        case context
        case resolvedSet = "resolved_set"
        case generatedAt = "generated_at"
    }
}

internal struct SdkFeedContextPayload: Codable, Equatable, Sendable {
    let platform: String
    let appVersion: String
    let userSegments: [String]

    enum CodingKeys: String, CodingKey {
        case platform
        case appVersion = "app_version"
        case userSegments = "user_segments"
    }
}

internal struct SdkFeedSetPayload: Codable, Equatable, Sendable {
    let id: String
    let revisionId: String
    let placementKey: String
    let isFallback: Bool
    let groups: [SdkFeedGroupPayload]

    enum CodingKeys: String, CodingKey {
        case id
        case revisionId = "revision_id"
        case placementKey = "placement_key"
        case isFallback = "is_fallback"
        case groups
    }
}

internal struct SdkFeedGroupPayload: Codable, Equatable, Sendable {
    let id: String
    let revisionId: String
    let title: String
    let bottomLabel: String?
    let logoURL: String
    let badge: SdkFeedBadgePayload?
    let stories: [SdkFeedStoryPayload]

    enum CodingKeys: String, CodingKey {
        case id
        case revisionId = "revision_id"
        case title
        case bottomLabel = "bottom_label"
        case logoURL = "logo_url"
        case badge
        case stories
    }
}

internal struct SdkFeedBadgePayload: Codable, Equatable, Sendable {
    let type: String
    let value: String
}

internal struct SdkFeedStoryPayload: Codable, Equatable, Sendable {
    let id: String
    let revisionId: String
    let title: String
    let mediaType: String
    let imageDurationMs: Int64?
    let asset: SdkFeedAssetPayload
    let posterAsset: SdkFeedAssetPayload?
    let cta: SdkFeedCtaPayload?

    enum CodingKeys: String, CodingKey {
        case id
        case revisionId = "revision_id"
        case title
        case mediaType = "media_type"
        case imageDurationMs = "image_duration_ms"
        case asset
        case posterAsset = "poster_asset"
        case cta
    }
}

internal struct SdkFeedAssetPayload: Codable, Equatable, Sendable {
    let id: String
    let url: String
    let mimeType: String
    let width: Int?
    let height: Int?
    let durationMs: Int64?

    enum CodingKeys: String, CodingKey {
        case id
        case url
        case mimeType = "mime_type"
        case width
        case height
        case durationMs = "duration_ms"
    }
}

internal struct SdkFeedCtaPayload: Codable, Equatable, Sendable {
    let label: String
    let type: String
    let value: String
}
